empress_gui.py

```diff
@@ -4,362 +4,819 @@
+        self.view_pvalue_histogram_btn.configure(state=tk.DISABLED)
 
-        dup_label = tk.Label(costs_frame, text="Duplication:")
-        self.dup_error = tk.Label(costs_frame, text="")
+        # Creates a new tkinter window 
+        self.cost_space_window = tk.Toplevel(self.master)
+        self.cost_space_window.geometry("550x550")
+        self.cost_space_window.title("Matplotlib Graph - Cost regions")
+        # Creates a new frame
+        plt_frame = tk.Frame(self.cost_space_window)
+        plt_frame.pack(fill=tk.BOTH, expand=1)
+        plt_frame.pack_propagate(False)
+        cost_regions = empress.compute_cost_regions(self.recon_input, 0.5, 10, 0.5, 10)  
+        fig = cost_regions.draw()  # creates matplotlib figure
+        canvas = FigureCanvasTkAgg(fig, plt_frame)
+        canvas.draw()
+        canvas.get_tk_widget().pack(side=tk.TOP, fill=tk.BOTH, expand=True)
+        # The toolbar allows the user to zoom in/out, drag the graph and save the graph
+        toolbar = NavigationToolbar2Tk(canvas, plt_frame)
+        toolbar.update()
+        canvas.get_tk_widget().pack(side=tk.TOP)
+        # Updates the DTL costs using the x,y coordinates clicked by the user inside the graph 
+        # Otherwise pops up a warning message window
+        fig.canvas.callbacks.connect('button_press_event', self.get_xy_coordinates)
+
+    def get_xy_coordinates(self, event):
+        """Update the DTL costs when user clicks on the matplotlib graph, otherwise pop up a warning message window."""
+        if event.inaxes is not None:
+            self.dup_input.set(round(event.xdata, 2))
+            self.trans_input.set(round(event.ydata, 2))
+            self.loss_input.set("1.00")
+            # Enables the next step, viewing reconciliation result
+            self.dup_cost = event.xdata
+            self.trans_cost = event.ydata
+            self.loss_cost = 1.00
+            self.update_compute_reconciliations_btn()
+        else:
+            messagebox.showinfo("Warning", "Please click inside the axes bounds.")
+
+    def dtl_cost(self):
+        """Set DTL costs by clicking on the matplotlib graph or by entering manually."""  
+        self.dup_label = tk.Label(self.costs_frame, text="Duplication cost:")
+        self.dup_label.grid(row=0, column=0, sticky="w")
         # %P = value of the entry if the edit is allowed
         # see https://stackoverflow.com/questions/4140437/interactively-validating-entry-widget-content-in-tkinter
         dup_vcmd = (self.register(self.validate_dup_input), '%P')
-        self.dup_input = tk.DoubleVar()
-        self.dup_entry_box = tk.Entry(costs_frame, width=3, validate="all", textvariable=self.dup_input, validatecommand=dup_vcmd)
-        
-        dup_label.grid(row=0, column=0, sticky="w")
+        # self.dup_input is tk.DoubleVar(), initialized to be 1.00
+        self.dup_entry_box = CustomEntry(self.costs_frame, width=3, textvariable=self.dup_input)
```

Comment:
I would prefer more width, but you should consult Prof Ran about this.

Reply:
Changed to width=4.